x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx

```diff
@@ -202,31 +225,48 @@ const AlertsTableState = ({
       pageSize: pagination.pageSize,
       pageSizeOptions: [10, 20, 50, 100],
       leadingControlColumns: [],
-      showCheckboxes,
       showExpandToDetails,
       trailingControlColumns: [],
       useFetchAlertsData,
       visibleColumns: storageAlertsTable.current.visibleColumns ?? [],
       'data-test-subj': 'internalAlertsState',
+      query,
     }),
     [
       alertsTableConfiguration,
       columns,
       flyoutSize,
       pagination.pageSize,
-      showCheckboxes,
       showExpandToDetails,
       useFetchAlertsData,
+      query,
     ]
   );
 
+  const CasesContext = cases?.ui.getCasesContext();
+
   return hasAlertsTableConfiguration ? (
     <>
       {!isLoading && alertsCount === 0 && <EmptyState />}
       {isLoading && (
         <EuiProgress size="xs" color="accent" data-test-subj="internalAlertsPageLoading" />
       )}
-      {alertsCount !== 0 && <AlertsTable {...tableProps} />}
+      {alertsCount !== 0 && CasesContext && cases && (
+        <CasesContext
+          owner={[configurationId]}
+          permissions={cases.permissions}
+          features={{ alerts: { sync: false } }}
+        >
+          <BulkActionsContext.Provider value={initialBulkActionsState}>
```

Comment:
nit -> To avoid duplicate code, I will put the code below in a `useMemo`
```
<BulkActionsContext.Provider value={initialBulkActionsState}>
   <AlertsTable {...tableProps} />
</BulkActionsContext.Provider>
```